### ISSUE TITLE
Help tools are misdetected on recent Qt version

### ISSUE THREAD

**Issue by krop** [KDAB/GammaRay#830]
qtattributionscanner and qhelpgenerator are libexec tools with recent Qt6 version.

This could fix it, I'll make a PR after testing it

```
diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index 20c663211..e448e76dc 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -25,13 +25,13 @@ if(TARGET Qt::qhelpgenerator)
     # Required for Doxygen
     get_target_property(QHELPGEN_EXECUTABLE Qt::qhelpgenerator IMPORTED_LOCATION)
 else()
-    find_program(QHELPGEN_EXECUTABLE qhelpgenerator HINTS ${QT_INSTALL_BINS})
+    find_program(QHELPGEN_EXECUTABLE qhelpgenerator HINTS ${QT_INSTALL_BINS} ${QT_INSTALL_LIBEXECS})
 endif()
 
 if(TARGET Qt::qtattributionsscanner)
     set(QTATTRIBUTIONSSCANNER_EXECUTABLE Qt::qtattributionsscanner)
 else()
-    find_program(QTATTRIBUTIONSSCANNER_EXECUTABLE qtattributionsscanner HINTS ${QT_INSTALL_BINS})
+    find_program(QTATTRIBUTIONSSCANNER_EXECUTABLE qtattributionsscanner HINTS ${QT_INSTALL_BINS} ${QT_INSTALL_LIBEXECS})
 endif()
```

**Comment by ferdnyc** [comment]
@krop Nice catch!

Though, I admit I'm really surprised that "recent Qt6 versions" would even be using the fallback `find_program()` scanning. Does your install not have the `Qt::qhelpgenerator` and `Qt::qtattributionsscanner` CMake targets available?

**Comment by krop** [comment]
> @krop Nice catch!
> 
> Though, I admit I'm really surprised that "recent Qt6 versions" would even be using the fallback `find_program()` scanning. Does your install not have the `Qt::qhelpgenerator` and `Qt::qtattributionsscanner` CMake targets available?

Check the pull request, there was another issue that has also been fixed.

**Comment by ferdnyc** [comment]
@krop
> Check the pull request, there was another issue that has also been fixed.

Yeah, I saw that part... I guess it's because the **very first** Qt probe GammaRay does is for "`QT`" in all caps:

https://github.com/KDAB/GammaRay/blob/93428a9da835c04162fa817678e44795d0f80033/CMakeLists.txt#L428-L436

Because, when I do this:

```cmake
find_package(Qt 5.5 NAMES Qt6 Qt5 COMPONENTS Core NO_MODULE REQUIRED)
```

...on Fedora 38, _only_ `${Qt_VERSION_MAJOR}` is defined. But when I do this:

```cmake
find_package(QT 5.5 NAMES Qt6 Qt5 COMPONENTS Core NO_MODULE REQUIRED)
```

_only_ `${QT_VERSION_MAJOR}` is defined. (Pretty strange of Qt to not just set both capitalizations, at least in the `find_package(Qt...)` scenario.)

**Comment by krop** [comment]
> @krop
> 
> > Check the pull request, there was another issue that has also been fixed.
> 
> Yeah, I saw that part... I guess it's because the **very first** Qt probe GammaRay does is for "`QT`" in all caps:
> 
> https://github.com/KDAB/GammaRay/blob/93428a9da835c04162fa817678e44795d0f80033/CMakeLists.txt#L428-L436
> 

That's wrong, indeed. But this change was pushed after the 3.0.0 release.

@winterz https://github.com/KDAB/GammaRay/commit/56cbf79e6 replaced 'Qt' with 'QT'

**Comment by krop** [comment]
> Since these commits are not part of the 3.0.0 release, can you create a new issue with your findings please? (I'll complete with mines, I see 3 different issues in this code)

Correction, they are. I'll submit the report.
